fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+
+        return self
+
+    def partial_fit(self, X, y, *, sensitive_features=None):
+        """
+        Perform one epoch on given samples and update model.
+
+        Parameters
+        ----------
+        X : numpy.ndarray
+            Two-dimensional numpy array containing training data
+
+        y : array
+            Array-like containing training targets
+
+        sensitive_features : array
+            Array-like containing the sensitive feature of the
+            training data.
+        """
+        X, Y, Z = self._validate_input(
+            X, y, sensitive_features, reinitialize=False
+        )
+        self.backendEngine_.train_step(X, Y, Z)
+
+        return self
+
+    def decision_function(self, X):
+        """
+        Compute predictor output for given test data.
+
+        Parameters
+        ----------
+        X : numpy.ndarray
+            Two-dimensional numpy array containing test data
+
+        Returns
+        -------
+        Y_pred : numpy.ndarray
+            Two-dimensional array containing the model's (soft-)predictions
+        """
+        check_is_fitted(self)
+        X = check_X(X)
+
+        Y_pred = self.backendEngine_.evaluate(X)
+        return Y_pred
+
+    def predict(self, X):
+        """
+        Compute discrete predictions for given test data.
+
+        Parameters
+        ----------
+        X : numpy.ndarray
+            Two-dimensional numpy array containing test data
+
+        Returns
+        -------
+        Y_pred : array
+            array-like containing the model's predictions fed through
+            the (discrete) :code:`predictor_function`
+        """
+        check_is_fitted(self)
+        X = check_X(X)
+
+        Y_pred = self.backendEngine_.evaluate(X)
+        Y_pred = self.predictor_function_(Y_pred)
+        Y_pred = self.y_transform_.inverse_transform(Y_pred)
+        return Y_pred
+
+    def _validate_input(self, X, Y, Z, reinitialize=False):
+        """
+        Validate the input data and possibly setup this estimator.
+
+        Important note is that we follow call `__setup` from here, because the
+        setup procedure requires the validated data. If `reintialize` is True,
+        then always call `__setup`.
+        """
+        if not self.skip_validation:
+            X = check_X(X)
+
+        try:  # TODO check this
+            check_is_fitted(self)
+            is_fitted = True
+        except NotFittedError:
+            is_fitted = False
+
+        if Z is None:
+            print("Warning: no sensitive_features provided")  # FIXME : logger?
+            Z = [0] * X.shape[0]
+
+        if (not is_fitted) or (reinitialize):
+            self.__setup(X, Y, Z)
+
+        if not self.skip_validation:
+            Y = self.y_transform_.transform(Y)
+            Z = self.z_transform_.transform(Z)
+
+        # Check for equal number of samples
+        if not (X.shape[0] == Y.shape[0] and X.shape[0] == Z.shape[0]):
```

Comment:
https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_consistent_length.html

Reply:
Even here, in case we are certain that everything is a numpy array?

Reply:
it's not about data conversion, it's about not writing the same code multiple times and having consistent error messages.